extensions/wikia/Email/Controller/FounderDigestController.class.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+use Email\Check;
+use Email\Tracking\TrackingCategories;
+
+abstract class FounderDigestController extends EmailController {
+	// Defaults; will be overridden in subclasses
+	const TRACKING_CATEGORY_EN = TrackingCategories::DEFAULT_CATEGORY;
+	const TRACKING_CATEGORY_INT = TrackingCategories::DEFAULT_CATEGORY;
+
+	protected $wikiId;
+	protected $pageViews;
+
+	public function initEmail() {
+		$this->wikiId = $this->request->getVal( 'wikiId' );
+		$this->pageViews = $this->request->getVal( 'pageViews' );
+		$this->assertValidParams();
+
+		$this->pageViews = number_format( $this->pageViews );
+	}
+
+	protected function getSummary() {
+		return $this->getMessage( 'emailext-founder-digest-summary',
+			\WikiMap::getWikiName( $this->wikiId ) )->parse();
```

Comment:
Ok, a few things here.  I think ```WikiMap``` is some kind of old MediaWiki attempt at hosting multiple wikis, and as such I don't think it works with our setup.  For getting wiki information you should use the methods in ```WikiFactory```.  The commonly seen ones are things like:

* WikiFactory::IDtoDB - maps a wiki ID to a dbname
* WikiFactory::DBtoDomain - maps a dbname to a domain name
* WikiFactory::DBtoUrl - I think you get the point ...

For this line however, there is no easy, single function to get the wiki title.  The best thing I think would be:
```php
$wikiObj = WikiFactory::getWikiByID( $this->wikiId );
$name = empty( $wikiObj ) ? '' : $wikiObj->city_title;
```

The second thing here is that ```emailext-founder-digest-summary``` does not have a ```$1``` to receive this value.  Its currently using ```{{SITENAME}}``` which will always resolve to the current wiki name.